libraries/botbuilder-ai/src/luis-client/luisClient.ts

```diff
@@ -0,0 +1,200 @@
+/**
+ * @module botbuilder-ai
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+/**
+ * NOTE: This class was auto generated by OpenAPI Generator (https://openapi-generator.tech)
+ * and was manually modified to make it compliant with the current implementation of the library.
+ */
+
+const fetch = (new Function('require', 'if (!this.hasOwnProperty("fetch")) { return require("node-fetch"); } else { return this.fetch; }'))(require);
+
+import * as HttpStatus from 'http-status-codes';
+
+/* tslint:disable:no-unused-locals */
+import { LuisResult } from './model/luisResult';
+
+import { ObjectSerializer, Authentication, VoidAuth } from './model/models';
+import { ApiKeyAuth } from './model/models';
+
+let luisVersion = '/luis/v2.0';
+
+export interface PredictionResolveOptionalParams {
+    /**
+     * The timezone offset for the location of the request.
+     */
+    timezoneOffset?: number;
+    /**
+     * If true, return all intents instead of just the top scoring intent.
+     */
+    verbose?: boolean;
+    /**
+     * Use the staging endpoint slot.
+     */
+    staging?: boolean;
+    /**
+     * Enable spell checking.
+     */
+    spellCheck?: boolean;
+    /**
+     * The subscription key to use when enabling bing spell check
+     */
+    bingSpellCheckSubscriptionKey?: string;
+    /**
+     * Log query (default is true)
+     */
+    log?: boolean;
+
+    customHeaders: {headers: {[name: string]: string}};
+}
+
+ /**
+  *  Sets headers for request to LUIS service.
+  *  LUIS service uses the `Ocp-Apim-Subscription-Key` header. 
+  */
+export enum LuisApikeys {
+    apiKeyHeader,
+}
+
+/**
+ * Luis Client Class
+ *
+ * @remarks
+ * This is a helper class for all the http request operations against the LUIS API.
+ * https://www.luis.ai/
+ */
+export class LuisClient {
+    /** Supported Cognitive Services endpoint (protocol and hostname, for example: https://westus.api.cognitive.microsoft.com) */
+    private _basePath: string = '';
+
+    protected authentications = {
+        'default': new VoidAuth() as Authentication,
+        'apiKeyHeader': new ApiKeyAuth('header', 'Ocp-Apim-Subscription-Key'),
+    }
+
+    /** Creates a new instance of LuisClient.
+     * @param basePath Supported Cognitive Services endpoint.
+     */
+    public constructor(basePath: string){
+        if (basePath) {
+            this._basePath = basePath + luisVersion;
+        }
+    }
```

Comment:
This logic doesn't quite support custom host names and endpoints. @chrimc62 what's the proper logic to use here?